### PR TITLE
RFE 657 : Realigning privileges after DB/Tbl/Col/Proc Rename/Copy/Move

### PR DESCRIPTION
**Pull request by devenbansod**
Adding an option to realign the database specific, table specific, column specific and procedures specific privileges after Copying / Renaming the database.

Signed-off-by: Deven Bansod devenbansod.bits@gmail.com

### CODE REVIEW

**Comment by devenbansod** [comment]
Work in Progress. Done only for renaming/ copying Databases. 


Also discussion is going on with respect to few doubts on the ticket.

**Comment by lem9** [comment]
I suggest rebasing your branch because of a recent fix in master.

**Comment by devenbansod** [comment]
Thanks @lem9 . Did the rebasing!

**Comment by ibennetch** [comment]
Great work, but there's one small issue. I think you need to reload (flush) the privileges after making the change. This can be done manually from https://github.com/phpmyadmin/phpmyadmin/blob/master/libraries/server_privileges.lib.php#L4246-L4264 and https://github.com/phpmyadmin/phpmyadmin/blob/master/libraries/server_privileges.lib.php#L3717-3722 but in this case we should do it automatically.

**Comment by devenbansod** [comment]
Yes. Flushing(reloading) should be automatic. I will run the 'FLUSH PRIVILEGES' sql query after updating all the privileges.

And I think whenever we have realigned the privileges, we should append to the original success message of 'Table Successfully Copied/ Moved' that 'The privileges have been realigned and the changes have been flushed'.

**Comment by ibennetch** [comment]
Good idea, but I don't think we need that level of detail -- "The privileges have been realigned" should be enough. I think the user assumes we'll also perform the flush in this case.

Can you also make the checkbox selected by default? I think this should be the default behavior when moving a table.

**Comment by devenbansod** [comment]
Yes. That seems a valid point too.

And yes, I will make it checked by default. It's right. The user should not have to specifically remember to realign the privileges. It should be done by default just that we provide him an option to choose not to do it.

**Comment by devenbansod** [comment]
Added the privileges option for Table's Copy/ Rename/ Move as well Columns Renaming. 

Please suggest changes.

File: tbl_operations.php [hunk 1]
```diff
@@ -109,7 +109,19 @@
     $warning_messages = array();
 
     if (isset($_REQUEST['new_name'])) {
+        // Get original names before rename operation
+        $oldTbl = $pma_table->getName();
```
Comment (ibennetch) [inline tbl_operations.php]: Is it a problem to use `$oldTable` here to be a bit more readable?

Comment (ibennetch) [inline tbl_operations.php]: I wonder if we should use `$oldTable` to make it a bit more readable. What do you think?

File: libraries/operations.lib.php [hunk 2]
```diff
@@ -1594,6 +1751,91 @@ function PMA_getQueryAndResultForPartition()
...
+ * Realign the privileges after renaming/moving a table
+ *
+ * @param string $oldDb  Database name before table renaming/moving table
+ * @param string $oldTbl Table name before table renaming/moving table
+ * @param string $newDb  Database name after table renaming/ moving table
+ * @param string $newTbl Table name after table renaming/moving table
+ *
+ * @return void
+ */
+function PMA_RealignPrivileges_renameOrMoveTbl($oldDb, $oldTbl, $newDb, $newTbl)
```
Comment (ibennetch) [inline libraries/operations.lib.php]: I wonder if using `PMA_RealignPrivileges_renameOrMoveTable` instead would improve readability here, too.

File: tbl_operations.php (hunk 1 above)
Comment (devenbansod) [inline tbl_operations.php]: Now that you pointed out, I agree that for tables, $table is far more clear than $tbl.

Also, it appears $table is the standard used in our code, too.

I will make the appropriate changes and commit back.

**Comment by ibennetch** [comment]
Looks good to me. Can you squash the commits after which I think it's ready for merge.

**Comment by devenbansod** [comment]
Squashed the commits.

**Comment by ibennetch** [comment]
Nice work Deven.

**Comment by devenbansod** [comment]
Thanks. :)

**Comment by devenbansod** [comment]
Please see #1683.

**Pull request merged**